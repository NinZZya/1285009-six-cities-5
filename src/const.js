export const AppPath = {
  ROOT: `/`,
  LOGIN: `/login`,
  CITY: `/city`,
  OFFER: `/offer`,
  FAVORITES: `/favorites`,
  NOT_FOUND: `/not-found`,
};

export const CITIES = {
  1: {id: 1, name: `Paris`},
  2: {id: 2, name: `Cologne`},
  3: {id: 3, name: `Brussels`},
  4: {id: 4, name: `Amsterdam`},
  5: {id: 5, name: `Hamburg`},
  6: {id: 6, name: `Dusseldorf`},
};

export const SortType = {
  POPULAR: `Popular`,
  LOW_TO_HIGH: `Price: low to high`,
  HIGH_TO_LOW: `Price: high to low`,
  TOP_RATED_FIRST: `Top rated first`,
};

export const STARS = [
  {value: 5, title: `perfect`},
  {value: 4, title: `good`},
  {value: 3, title: `not bad`},
  {value: 2, title: `badly`},
  {value: 1, title: `terribly`},
];
