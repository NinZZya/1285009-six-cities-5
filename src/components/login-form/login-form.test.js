import React from 'react';
import renderer from 'react-test-renderer';
import LoginForm from './login-form';
import {UserStatus} from '~/constants/const';


const ERROR = `Error`;
const testing = () => {};

describe(`Should LoginForm render correctly`, () => {
  it(`Should LoginForm render correctly no auth`, () => {
    const tree = renderer
      .create(
          <LoginForm
            userStatus={UserStatus.NO_AUTH}
            onAuth={testing}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should LoginForm render correctly error`, () => {
    const tree = renderer
      .create(
          <LoginForm
            userStatus={UserStatus.ERROR}
            onAuth={testing}
            error={ERROR}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should LoginForm render correctly responce`, () => {
    const tree = renderer
      .create(
          <LoginForm
            userStatus={UserStatus.RESPONSE}
            onAuth={testing}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
