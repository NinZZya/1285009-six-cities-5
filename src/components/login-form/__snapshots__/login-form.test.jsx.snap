// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should LoginForm render correctly Should LoginForm render correctly error 1`] = `
<form
  action="#"
  className="login__form form"
  method="post"
  onSubmit={[Function]}
>
  <div
    className="login__input-wrapper form__input-wrapper"
  >
    <label
      className="visually-hidden"
    >
      E-mail
    </label>
    <input
      className="login__input form__input"
      disabled={false}
      name="email"
      placeholder="Email"
      required=""
      type="email"
    />
  </div>
  <div
    className="login__input-wrapper form__input-wrapper"
  >
    <label
      className="visually-hidden"
    >
      Password
    </label>
    <input
      className="login__input form__input"
      disabled={false}
      name="password"
      placeholder="Password"
      required=""
      type="password"
    />
  </div>
  <button
    className="login__submit form__submit button"
    disabled={false}
    type="submit"
  >
    Sign in
  </button>
</form>
`;

exports[`Should LoginForm render correctly Should LoginForm render correctly no auth 1`] = `
<form
  action="#"
  className="login__form form"
  method="post"
  onSubmit={[Function]}
>
  <div
    className="login__input-wrapper form__input-wrapper"
  >
    <label
      className="visually-hidden"
    >
      E-mail
    </label>
    <input
      className="login__input form__input"
      disabled={false}
      name="email"
      placeholder="Email"
      required=""
      type="email"
    />
  </div>
  <div
    className="login__input-wrapper form__input-wrapper"
  >
    <label
      className="visually-hidden"
    >
      Password
    </label>
    <input
      className="login__input form__input"
      disabled={false}
      name="password"
      placeholder="Password"
      required=""
      type="password"
    />
  </div>
  <button
    className="login__submit form__submit button"
    disabled={false}
    type="submit"
  >
    Sign in
  </button>
</form>
`;

exports[`Should LoginForm render correctly Should LoginForm render correctly responce 1`] = `
<form
  action="#"
  className="login__form form"
  method="post"
  onSubmit={[Function]}
>
  <div
    className="login__input-wrapper form__input-wrapper"
  >
    <label
      className="visually-hidden"
    >
      E-mail
    </label>
    <input
      className="login__input form__input"
      disabled={true}
      name="email"
      placeholder="Email"
      required=""
      type="email"
    />
  </div>
  <div
    className="login__input-wrapper form__input-wrapper"
  >
    <label
      className="visually-hidden"
    >
      Password
    </label>
    <input
      className="login__input form__input"
      disabled={true}
      name="password"
      placeholder="Password"
      required=""
      type="password"
    />
  </div>
  <button
    className="login__submit form__submit button"
    disabled={true}
    type="submit"
  >
    Sign in
  </button>
</form>
`;
